src/validation.cpp

```diff
@@ -1852,6 +1852,20 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
         flags |= SCRIPT_VERIFY_NULLDUMMY;
     }
 
+    // BIP148 mandatory segwit signalling.
+    int64_t nMedianTimePast = pindex->GetMedianTimePast();
+    if ( (nMedianTimePast >= 1501545600) &&  // Tue 01 Aug 2017 00:00:00 UTC
```

Comment:
Good flag day would be July 1st 2017. That gives 4 weeks of experience before Bitcoin BIP148 and will get a lot of support from Bitcoin supporters as well.